internal/config/config.go

```diff
@@ -12,10 +17,39 @@ type Config struct {
 var C *Config
 
 func init() {
-	C = &Config{
-		DefaultWindowSize: r2.Vec{X: 1280, Y: 720},
-		Ratio:             1280.0 / 720.0,
-		TargetTPS:         60,
-		Debug:             true,
+	configstore.RegisterProvider("defaults", getDefaultConfiguration)
+	configstore.File("configuration.yml")
+
+	items, err := configstore.Filter().Squash().GetItemList()
+	if err != nil {
+		panic(err)
+	}
+
+	x, _ := items.GetItemValueFloat("DefaultWindowSize.X")
```

Comment:
I think what we want to end up with is a struct with all our config, preferably by just unmarshalling, like in https://github.com/knadh/koanf?tab=readme-ov-file#unmarshalling-and-marshalling